### PR TITLE
DTSPB-1161: Setup chrome crossbrowser tests in nightly pipeline

### PR DESCRIPTION
**Pull request by Rob-Doherty**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-1161

### Change description ###

This change adds the crossbrowser tests to the nightly pipeline, which will mean the Chrome browser tests run overnight.
It includes 
- update of Jenkinsfile_nightly to turn on crossbrowser tests
- update of test runner to handle upcoming parallel Jenkins steps
- some improvements to checks and waiting on pages to load fully before continuing
- a lot of `await`s to make sure that the test steps run synchronously as a real user would

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by Rob-Doherty** [comment]
These changes are working in the nightly pipeline: 
image

**Comment by Rob-Doherty** [comment]
These changes work as expected in the nightly pipeline:
https://build.platform.hmcts.net/job/HMCTS_Nightly_Probate/job/probate-caveats-frontend/view/change-requests/job/PR-346/
image

**Review by mounikahmcts — APPROVED** [review]

File: test/end-to-end/helpers/JSWait.js [hunk 1]
```diff
@@ -14,13 +14,13 @@ class JSWait extends codecept_helper {
         const helper = this.helpers.WebDriver || this.helpers.Puppeteer;
         const helperIsPuppeteer = this.helpers.Puppeteer;
 
-        helper.click(text, locator).catch(err => {
-            console.error(err.message);
-        });
-
         if (helperIsPuppeteer) {
+            helper.click(text, locator).catch(err => {
```
Comment (vinceholt) [inline test/end-to-end/helpers/JSWait.js:18]: best to use await and a try/catch block here
so, move this before the if:
try {
await helper.click(text, locator);
} catch (err) {
console.error(err.message);
}

File: test/end-to-end/pages/deceased/alias.js [hunk 2]
```diff
@@ -9,8 +9,10 @@ async function enterDeceasedHasAlias(language='en') {
     const commonContent = language === 'en' ? commonContentEn : commonContentCy;
     const I = this;
 
-    I.seeInCurrentUrl(pageUnderTest.getUrl());
-    await I.click('#alias');
+    I.waitInUrl(pageUnderTest.getUrl());
```
Comment (vinceholt) [inline test/end-to-end/pages/deceased/alias.js:12]: add an await here

File: test/end-to-end/helpers/JSWait.js (hunk 1 above)
Comment (Rob-Doherty) [reply test/end-to-end/helpers/JSWait.js:18]: good thought, will add that change

File: test/end-to-end/pages/deceased/alias.js (hunk 2 above)
Comment (Rob-Doherty) [reply test/end-to-end/pages/deceased/alias.js:12]: good spot!! I thought I'd whacked 'em all 😄

File: test/end-to-end/helpers/JSWait.js (hunk 1 above)
Comment (Rob-Doherty) [reply test/end-to-end/helpers/JSWait.js]: done

File: test/end-to-end/pages/deceased/alias.js (hunk 2 above)
Comment (Rob-Doherty) [reply test/end-to-end/pages/deceased/alias.js]: done

**Review by vinceholt — APPROVED** [review]

**Pull request merged**